src/integration-test/java/com/commercetools/sync/integration/sdk2/externalsource/states/StateSyncIT.java

```diff
@@ -0,0 +1,1285 @@
+            .roles(List.of(StateRoleEnum.REVIEW_INCLUDED_IN_STATISTICS))
+            .build();
+
+    final StateSyncOptions stateSyncOptions = StateSyncOptionsBuilder.of(CTP_TARGET_CLIENT).build();
+
+    final StateSync stateSync = new StateSync(stateSyncOptions);
+
+    // test
+    final StateSyncStatistics stateSyncStatistics =
+        stateSync.sync(List.of(stateDraft)).toCompletableFuture().join();
+
+    assertThat(stateSyncStatistics).hasValues(1, 1, 0, 0, 0);
+  }
+
+  @Test
+  void sync_withCreateStateException_shouldPrintMessage() {
+    final StateDraft stateDraft =
+        StateDraftBuilder.of()
+            .key(keyA)
+            .type(StateTypeEnum.REVIEW_STATE)
+            .roles(List.of(StateRoleEnum.REVIEW_INCLUDED_IN_STATISTICS))
+            .build();
+
+    final ProjectApiRoot client = buildClientWithBadGatewayExceptionOnPost();
+
+    final StateSyncOptions stateSyncOptions =
+        StateSyncOptionsBuilder.of(client)
+            .errorCallback(
+                (exception, oldResource, newResource, updateActions) -> {
+                  errorCallBackMessages.add(exception.getMessage());
+                  errorCallBackExceptions.add(exception.getCause());
+                })
+            .build();
+
+    final StateSync stateSync = new StateSync(stateSyncOptions);
+
+    // test
+    final StateSyncStatistics stateSyncStatistics =
+        stateSync.sync(List.of(stateDraft)).toCompletableFuture().join();
+
+    assertThat(stateSyncStatistics).hasValues(1, 0, 0, 1, 0);
+    assertThat(errorCallBackExceptions).isNotEmpty();
+    assertThat(errorCallBackMessages).isNotEmpty();
+    assertThat(errorCallBackMessages.get(0))
+        .contains(format("Failed to create draft with key: '%s'", keyA));
+  }
+
+  @Test
+  void sync_withNewStateWithoutRole_shouldRemoveRole() {
+    final StateDraft stateDraft =
+        StateDraftBuilder.of().key(keyA).type(StateTypeEnum.REVIEW_STATE).roles(List.of()).build();
+
+    final StateDraft stateDraftTarget =
+        StateDraftBuilder.of()
+            .key(keyA)
+            .type(StateTypeEnum.REVIEW_STATE)
+            .roles(List.of(StateRoleEnum.REVIEW_INCLUDED_IN_STATISTICS))
+            .build();
+    createStateInTarget(stateDraftTarget);
+
+    final StateSyncOptions stateSyncOptions = StateSyncOptionsBuilder.of(CTP_TARGET_CLIENT).build();
+
+    final StateSync stateSync = new StateSync(stateSyncOptions);
+
+    // test
+    final StateSyncStatistics stateSyncStatistics =
+        stateSync.sync(List.of(stateDraft)).toCompletableFuture().join();
+
+    assertThat(stateSyncStatistics).hasValues(1, 0, 1, 0, 0);
+
+    final State fetchedState =
+        CTP_TARGET_CLIENT
+            .states()
+            .withKey(keyA)
+            .get()
+            .execute()
+            .thenApply(ApiHttpResponse::getBody)
+            .join();
+
+    assertThat(fetchedState.getRoles()).isEmpty();
+  }
+
+  @Test
+  void sync_withNewStateWithoutRole_shouldDoNothing() {
```

Comment:
the name is missleading, the test above is named `sync_withNewStateWithoutRole_shouldRemoveRole`, which describes the same setup with different outcome. Please adjust.

Reply:
https://github.com/commercetools/commercetools-sync-java/pull/1043/commits/19d7c3fd5dcc5e0058a8e4e9860092feb5b0b248